Libplanet/Net/Swarm.cs

```diff
@@ -1896,14 +1896,24 @@ await foreach (Block<T> block in blocks)
                     }
 
                     receivedBlockCount += receivedBlockCountCurrentLoop;
+                    var isEndedFirstTime = receivedBlockCount == receivedBlockCountCurrentLoop &&
+                                           receivedBlockCount < FindNextHashesChunkSize - 1;
 
-                    // FIXME: Need test
                     if (receivedBlockCountCurrentLoop < FindNextHashesChunkSize)
                     {
+                        if (!isEndedFirstTime)
+                        {
+                            continue;
+                        }
+
```

Comment:
```suggestion
                    if (receivedBlockCountCurrentLoop < FindNextHashesChunkSize && isEndedFirstTime)
                    {
```

Reply:
Also, I guess we can tweak `isEndedFirstTime` because `receivedBlockCount < FindNextHashesChunkSize - 1` and `receivedBlockCountCurrentLoop < FindNextHashesChunkSize` are redundant when `receivedBlockCount == receivedBlockCountCurrentLoop`

Reply:
fixed at https://github.com/planetarium/libplanet/pull/1115/commits/8df6fa762599550dab142f4f8c4687000451a806